presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -34,11 +34,11 @@ private BlockUtil()
     {
     }
 
-    static void checkValidPositionsArray(int[] positions, int offset, int length)
+    static void checkValidArrayRange(int[] array, int offset, int length)
```

Comment:
Why not just `checkArrayRange`? A check already implies that we are checking that the array range is valid.

Maybe we should rename similarly to Guava, they have a similar method named `checkPositionIndexes`.

Reply:
Good point! I think in our everything-is-a-position world, `checkArrayRange` provides less ambiguity, I'll go with that.